services/embedded-resource-updater.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+var P = require('bluebird');
+var mongoose = require('mongoose');
+var createError = require('http-errors');
+
+function EmbeddedResourceUpdater(model, params, record) {
+  this.perform = function () {
+    return new P(function (resolve, reject) {
```

Comment:
Is the promise really necessary?

Everything looks synchronous, except the `model.update`. So you just have to return it `return model.update(...)`.

Reply:
Done